### ISSUE TITLE
feature request: plenv local::lib support

### ISSUE THREAD

**Issue by inphobia** [Camelcade/Perl5-IDEA#2054]
hello, first up thx for fixing #2044 so quickly.

i do have a feature request for the wsl+plenv support should you have some spare time. while plenv itself does not seem to support local::lib they do reference https://github.com/miyagawa/plenv-contrib which adds this feature.

perhaps plenv and local::lib is a bit overkill but some of the code i'm working on really prefers to use it. (`use local::lib '/home/user/perl5';` without any fallbacks.).

so i was wondering if this could be considered to be added?

i tested on:
```
windows 10 1809

IntelliJ IDEA 2019.1.2 (Ultimate Edition)
Build #IU-191.7141.44, built on May 7, 2019
Licensed to netdisco and snmp::info / nick nauwelaerts
Subscription is active until May 6, 2020
For non-commercial open source development only.
JRE: 11.0.2+9-b159.56 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

perl5, perl5.lang.tt & perl5.lang.wsl 2019.1.2
```

```
windows 10 1903

IntelliJ IDEA 2019.2 EAP (Ultimate Edition)
Build #IU-192.4205.41, built on May 23, 2019
IntelliJ IDEA EAP User
Expiration date: June 22, 2019
Runtime version: 11.0.3+12-b248.2 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 2983M
Cores: 8
Registry: 

Non-Bundled Plugins: com.perl5, com.perl5.lang.tt2, com.perl5.transport.wsl
plugin versions "192.4205-EAP-CANDIDATE-SNAPSHOT"

java version "1.8.0_212"
Java(TM) SE Runtime Environment (build 1.8.0_212-b10)
Java HotSpot(TM) 64-Bit Server VM (build 25.212-b10, mixed mode)
```

both setups detect my wsl distributions just fine, as well as all install plenv versions. iirc the perl5 plugin uses `plenv versions` to detect available options. regretfully the local::lib plugin doesn't integrate with that, it adds 2 new commands to plenv:

console copy:
```
[ ~ ] % plenv help lib
Usage: plenv lib list
       plenv lib create [perl-version@]name
       plenv lib delete [perl-version@]name

Manages local::lib paths, separately to each installed perl.
When no 'perl-version@' part is given the current perl is assumed.

[ ~ ] % plenv help use
Usage: plenv use perl-version@lib
       plenv use perl-version
       plenv use @lib

Launches a new shell with specified perl version. When the '@lib' part is
specified, the local::lib name created with plenv lib is also activated.

When the perl version is empty, the current perl version is used.

[ ~ ] % plenv lib
5.28.1@netdisco-deploy
```

to make it ever more complicated you must actually start a subshell to make use of this all:
```
[ ~ ] % plenv use 5.28.1@netdisco-deploy

A sub-shell is launched with PLENV_VERSION=5.28.1 and local::lib activated for @netdisco-deploy. Run 'exit' to finish it.
```

bottom line, it would be much appreciated if this could somehow be added, but it seems none trivial.

thx for considering.

**Comment by hurricup** [comment]
There is a thing I don't really like with this `local::lib` support: it requires perl. When one of the feature of `plenv` is - you don't need to have a perl in the system. And this become even less tempting in the light of: https://developer.apple.com/documentation/macos_release_notes/macos_10_15_beta_release_notes#3318257

**Comment by hurricup** [comment]
Looks like it's not possible to integrate for now. See https://github.com/miyagawa/plenv-contrib/issues/12

**Comment by inphobia** [comment]
yup, i see the point. thx for checking but mostly thank you for making this awesome plugin.

**Comment by hurricup** [comment]
Don't close, things may change.
